tests/components/lock/test_init.py

```diff
@@ -2,328 +2,261 @@
 
 
+@pytest.mark.parametrize(("default_code", "code_format"), [("1234", r"^\d{4}$")])
 async def test_default_code_option_update(
     hass: HomeAssistant,
     entity_registry: er.EntityRegistry,
-    enable_custom_integrations: None,
+    mock_lock: MockLock,
 ) -> None:
     """Test default code stored in the registry is updated."""
 
-    entry = entity_registry.async_get_or_create("lock", "test", "very_unique")
-    await hass.async_block_till_done()
-
-    platform = getattr(hass.components, "test.lock")
-    platform.init(empty=True)
-
-    # Pre-register entities
-    entry = entity_registry.async_get_or_create("lock", "test", "very_unique")
     entity_registry.async_update_entity_options(
-        entry.entity_id,
-        "lock",
-        {
-            "default_code": "5432",
-        },
-    )
-    platform.ENTITIES["lock1"] = platform.MockLock(
-        name="Test",
-        code_format=r"^\d{4}$",
-        supported_features=LockEntityFeature.OPEN,
-        unique_id="very_unique",
+        "lock.test_lock", "lock", {CONF_DEFAULT_CODE: "4321"}
     )
-
-    assert await async_setup_component(hass, "lock", {"lock": {"platform": "test"}})
     await hass.async_block_till_done()
 
-    entity0: MockLock = platform.ENTITIES["lock1"]
-    assert entity0._lock_option_default_code == "5432"
-
-    entity_registry.async_update_entity_options(
-        entry.entity_id, "lock", {CONF_DEFAULT_CODE: "1234"}
-    )
-    await hass.async_block_till_done()
-
-    assert entity0._lock_option_default_code == "1234"
+    assert mock_lock._lock_option_default_code == "4321"
 
 
-async def test_lock_open_with_code(hass: HomeAssistant) -> None:
+@pytest.mark.parametrize(
+    ("code_format", "supported_features"),
+    [(r"^\d{4}$", LockEntityFeature.OPEN)],
+)
+async def test_lock_open_with_code(hass: HomeAssistant, mock_lock: MockLock) -> None:
     """Test lock entity with open service."""
-    lock = MockLockEntity(
-        code_format=r"^\d{4}$", supported_features=LockEntityFeature.OPEN
-    )
-    lock.hass = hass
-
-    assert lock.state_attributes == {"code_format": r"^\d{4}$"}
+    state = hass.states.get(TEST_ENTITY_ID)
+    assert state.attributes["code_format"] == r"^\d{4}$"
 
     with pytest.raises(ValueError):
-        await lock.async_handle_open_service()
+        await help_test_async_lock_service(hass, SERVICE_OPEN)
     with pytest.raises(ValueError):
-        await lock.async_handle_open_service(code="")
+        await help_test_async_lock_service(hass, SERVICE_OPEN, code="")
     with pytest.raises(ValueError):
-        await lock.async_handle_open_service(code="HELLO")
-    await lock.async_handle_open_service(code="1234")
-    assert lock.calls_open.call_count == 1
+        await help_test_async_lock_service(hass, SERVICE_OPEN, code="HELLO")
+    await help_test_async_lock_service(hass, SERVICE_OPEN, code="1234")
+    assert mock_lock.calls_open.call_count == 1
+    mock_lock.calls_open.assert_called_with(code="1234")
```

Comment:
To me, these are 4 different tests, with the 3 failing tests candidate to the use of `parametrize`.

The code checking on `code_format` above these would be yet another test.

This comment applies to the rest of the tests.

Reply:
The tests have in common that they test the service to open a lock. They prove the lock is only opened under expected conditions. Also, I try not to change the tests them selves, only the use of the entity platform.

Reply:
understood => limiting PR to the use of the entity platform.